Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,442 @@
+    printf("-------------------\n\r");
+}
+#endif
+
+/**
+ * @brief Reads inputs from the system
+*/
+void readInputs(){
+    #ifndef __TEST_SENDTRITIUM
+    Minion_Error_t err;
+
+    brakePedalPercent = Pedals_Read(BRAKE);
+    accelPedalPercent = Pedals_Read(ACCELERATOR);
+    
+    // UPDATE BUTTONS
+    if(Minion_Read_Input(REGEN_SW, &err) && regenCounter < DEBOUNCE_PERIOD){regenCounter++;}
+    else if(regenCounter > 0){regenCounter--;}
+
+    if(Minion_Read_Input(CRUZ_EN, &err) && cruiseEnableCounter < DEBOUNCE_PERIOD){cruiseEnableCounter++;}
+    else if(cruiseEnableCounter > 0){cruiseEnableCounter--;}
+
+    if(Minion_Read_Input(CRUZ_ST, &err) && cruiseSetCounter < DEBOUNCE_PERIOD){cruiseSetCounter++;}
+    else if(cruiseSetCounter > 0){cruiseSetCounter--;}
+    
+    // UPDATE GEARS
+    bool forwardSwitch = Minion_Read_Input(FOR_SW, &err);
+    bool reverseSwitch = Minion_Read_Input(REV_SW, &err);
+    forwardGear = (forwardSwitch && !reverseSwitch);
+    reverseGear = (!forwardSwitch && reverseSwitch);
+    neutralGear = (!forwardSwitch && !reverseSwitch);
+
+    if(forwardSwitch){
+        cruiseSet = Minion_Read_Input(CRUZ_ST, &err);
+    }
+    else{
+        cruiseEnable = false;
+        cruiseSet = false;
+        regenToggle = false;
+    }
+
+    if(forwardGear ^ reverseGear ^ neutralGear == false)
+    ; // TODO: jump to fault state for incorrect button read
+
+    // DEBOUNCING
+    if(regenCounter == DEBOUNCE_PERIOD){regenButton = true;}
+    else if(regenCounter == 0){regenButton = false;}
+
+    if(cruiseEnableCounter == DEBOUNCE_PERIOD){cruiseEnableButton = true;}
+    else if(cruiseEnableCounter == 0){cruiseEnableButton = false;}
+
+    if(cruiseSetCounter == DEBOUNCE_PERIOD){cruiseSet = true;}
+    else if(cruiseSetCounter == 0){cruiseSet = false;}
+
+    // TOGGLE
+    if(ChargeEnable && regenButton != regenPrevious && regenPrevious){regenToggle = !regenToggle;}
+    regenPrevious = regenButton;
+    
+    if(cruiseEnableButton != cruiseEnablePrevious && cruiseEnablePrevious){cruiseEnable = !cruiseEnable;}
+    cruiseEnablePrevious = cruiseEnableButton;
+
+    #endif
+}
+
+/**
+ * @brief Converts integer percentage to float percentage
+ * @param percent integer percentage from 0-100
+*/
+extern const float pedalToPercent[];
+static float percentToFloat(uint8_t percent){
+    if(percent > 100){
+        return 1.0f;
+    }
+    return pedalToPercent[percent];
+}
+
+/**
+ * @brief Follows the FSM to update the velocity of the car
+*/
+void Task_SendTritium(void *p_arg){
+    OS_ERR err;
+    Minion_Error_t minion_err;
+
+    // Initialize current state to NORMAL_DRIVE
+    state = FSM[NORMAL_DRIVE];
+
+    #ifndef __TEST_SENDTRITIUM
+    CANPayload_t velocityPayload;
+    velocityPayload.bytes = 32;
+    velocityPayload.data.d = velocitySetpoint;
+
+    CANPayload_t currentPayload;
+    currentPayload.bytes = 32;
+    currentPayload.data.d = currentSetpoint;
+    #endif
```

Comment:
This is using the old can driver. I've merged in the new one, but this code needs to be refactored to utilize it. Won't compile otherwise.